sale_timesheet_limit_date/models/sale_order_line.py

```diff
@@ -0,0 +1,39 @@
+# Copyright 2019 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo import api, models
+from odoo.exceptions import UserError
+from odoo.osv import expression
+
+
+class SaleOrderLine(models.Model):
+    _inherit = 'sale.order.line'
+
+    def _timesheet_compute_delivered_quantity_domain(self):
+        domain = super()._timesheet_compute_delivered_quantity_domain()
+        order = self.mapped('order_id')
+        if len(order) == 1:
+            if order.timesheet_limit_date:
+                domain = expression.AND([
+                    domain,
+                    [('date', '<=', self.order_id.timesheet_limit_date)]]
```

Comment:
As `self` can include multiple lines, this would raise a Singleton exception error.
```suggestion
                    [('date', '<=', order.timesheet_limit_date)]]
```